### DISCUSSION TITLE
Use a different composer executable

### DISCUSSION THREAD

**Discussion by NorthNick** [symfony/symfony#44716] [Q&A] [answered]
I've just upgraded a Symfony project to 6.0 and everything works fine locally. However it fails when deployed to the ISP's shared cloud server as their version of composer is 1.10 and symfony/framework-bundle requires composer-runtime-api >= 2.1. The machine has Composer 2.1.3 available as /opt/php/bin/composer2 so is there any way of making Symfony use that version?

I might create a local directory with a symbolic link to the new executable and prepend it to the PATH, but that's liable to be forgotten if we move to a new instance of the environment so I'm hoping there is a more direct way to change it in the project files somewhere.

**Comment by NorthNick** [comment, marked as answer]
Answering my own question...

Just installing the PHP packages using /opt/php/bin/composer2 does the trick. Embarrassingly simple.

**Reply by GromNaN** [reply]
For some cases, the var `COMPOSER_BINARY` must be set (example for `simple-phpunit`)

**Reply by NorthNick** [reply]
Thanks - that is useful to know.
